src/main.ts

```diff
@@ -482,6 +482,26 @@ export async function readPipeline(config: Config): Promise<string> {
   const filename = path.join(folderName, config.pipeline)
   core.debug(`Reading pipeline file from ${filename}`)
   let pipeline = fs.readFileSync(filename).toString()
+  const pipelineJson = JSON.parse(pipeline)
+  const runtimeParametersFile = pipelineJson.phases.verify?.context.runtime_parameters_file
+  if (runtimeParametersFile !== undefined) {
+    const runtimeParametersFilePath = path.join(folderName, runtimeParametersFile)
+    let runtimeParameters = Buffer.from(fs.readFileSync(runtimeParametersFilePath).toString().trim()).toString("base64")
+    // It's necesarry add the pading https://linuxhint.com/understand-base64-padding/
+    const runtimeParametersPadding = runtimeParameters.length % 4
```

Comment:
It would be good if you can extract the base64 encoding and base logic to a method and then that way you can write a couple of tests proving how it works. Which will be good internal docs too for the future.

Reply:
I think that actually, all the logic about reading the file, parsing it and injecting the parameters needs to go into a separate method so it can be well tested too.